### PR TITLE
Weather

### PR DESCRIPTION
**Pull request by macbright**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The project is part of a series of projects to be completed by students of [Micr
 - webpack
 
 ## Demo
- Check out the live demo of the project. [Click]()
+ Check out the live demo of the project. [Click](https://raw.githack.com/macbright/weather-app/weather/dist/index.html)
 
 ## Usage
```
Comment (SunnyAsar) [inline README.md]: Please update the README, the information here doesn't match the project.
- also is npm really needed for this project ?

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,152 @@
...
+};
+
+$('#submit').addEventListener('click', (e) => {
+	e.preventDefault();
+	const url = getApi();
+	fetch(url, { mode: 'cors' })
+		.then(response => response.json())
+		.then((response) => {
+			localStorage.setItem('city', JSON.stringify(response));
+			display();
```
Comment (SunnyAsar) [inline src/index.js]: why localStorage? we should be calling `display` with the data passed to it.

File: dist/index.html [hunk 3]
```diff
@@ -265,7 +266,30 @@
...
+				<div class="col-md-3 sunset ">
+					
+				</div>
+			</div>
+			<div class="row datas">
+				<div class="col-md-3 sys"></div>
+				<div class="col-md-4 main-data"></div>
+				<div class="col-md-4 weather"></div>
+				
+			</div>
```
Comment (SunnyAsar) [inline dist/index.html]: let's hide this section till there is weather data to be displayed.

File: src/index.js [hunk 4]
```diff
@@ -1,4 +1,152 @@
...
+		}).catch(err => `Error Reading data  ${err}`);
```
Comment (SunnyAsar) [inline src/index.js]: let's do better error handling to let the user know there is a problem. e.g when the city name does not exit.

File: src/index.js [hunk 5]
```diff
@@ -1,4 +1,152 @@
...
+const display = () => {
+	clearNodes();
+	changeBg();
+	const weatherData = JSON.parse(localStorage.getItem('city'));
+	cityname(weatherData.name, weatherData.weather[0].icon, weatherData.timezone);
+	degrees(weatherData.main.temp, weatherData.weather[0].icon);
+	latitute(weatherData.coord.lon, weatherData.coord.lat);
+	sys(weatherData.sys, weatherData.weather[0].icon);
+	mainData(weatherData.main, weatherData.weather[0].icon);
+	weather(weatherData.weather[0], weatherData.weather[0].icon);
```
Comment (SunnyAsar) [inline src/index.js]: let's use object de structuring as much as possible to pull out data from objects.

File: src/index.js [hunk 6]
```diff
@@ -1,4 +1,152 @@
...
+	} else if (sky === 'Clouds') {
+		$('body').style.backgroundImage = 'url(../dist/img/cloudy.gif)';
+	} else {
+		$('body').style.backgroundImage = 'url(../dist/img/weather.jpg)';
+	}
+};
+
+const getApi = () => {
+	const city = $('#city-name').value;
+	const country = $('#country').value;
```
Comment (SunnyAsar) [inline src/index.js]: I don't see the effect of the country. no matter the country i pick, the city result is returned. example country set to `spain` and city  `abuja` still return the data for abuja.
- If you like take out the country completely.

File: src/index.js [hunk 7]
```diff
@@ -1,4 +1,152 @@
...
+const $ = document.querySelector.bind(document);
 
-$('h1').style.color = 'blue';
\ No newline at end of file
+
+const convertToCelsius = (k) => {
+	const value = parseFloat((k - 273.15).toFixed(6));
+	return value;
+};
+
```
Comment (SunnyAsar) [inline src/index.js]: let's add a way on the UI to toggle between celsius to fahrenheit.

File: README.md (hunk 1 above)
Comment (macbright) [reply README.md]: i used webpack that's why included npm on the readme

File: src/index.js (hunk 2 above)
Comment (macbright) [reply src/index.js]: solved

File: dist/index.html (hunk 3 above)
Comment (macbright) [reply dist/index.html]: done

File: src/index.js (hunk 4 above)
Comment (macbright) [reply src/index.js]: done

File: src/index.js (hunk 5 above)
Comment (macbright) [reply src/index.js]: if i use destructing i will have to write more codes.

File: src/index.js (hunk 7 above)
Comment (macbright) [reply src/index.js]: done

**Pull request merged**